builtin/providers/cloudstack/resource_cloudstack_security_group.go

```diff
@@ -0,0 +1,210 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackSecurityGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackSecurityGroupCreate,
+		Read:   resourceCloudStackSecurityGroupRead,
+		Delete: resourceCloudStackSecurityGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"ingress_rules": &schema.Schema{
```

Comment:
Wouldn't it be easier to call this just `rules` and then have a parameter in the `schema` called `traffic_type`?

Have a look at this one as an example: https://github.com/hashicorp/terraform/blob/master/builtin/providers/cloudstack/resource_cloudstack_network_acl_rule.go

This way we can have both `ingress` and `egress` rules making sure people can configure what they need.